### DISCUSSION TITLE
Add "media" slot to element cards

### DISCUSSION THREAD

**Discussion by ryanleichty** [craftcms/cms#15651] [Ideas] [answered]
It feels inconsistent when showing a column of element cards where some have images and others don't. It would be great if element cards had the concept of a _media_ slot that could be optionally configured to display either a selected thumbnail image **or** fallback to something else (like the entry icon or something else).

![image](https://github.com/user-attachments/assets/e6c9f7c8-2e81-4bb3-bc3f-ed5233184b44)

Document previews in Sanity have the `title`, `subtitle`, and `media` slots that can be customized to display different values. In the screenshot below, the `media` slot looks for the first image within the `imageGallery` array field. If none exist, it falls back to the document icon.

![CleanShot 2024-09-03 at 16 00 59@2x](https://github.com/user-attachments/assets/b10b0361-cc1c-4a6a-b807-fe3d098140cb)

**Comment by ryanleichty** [comment]
This might be somewhat related to this PR: https://github.com/craftcms/cms/pull/15283

**Comment by brandonkelly** [comment]
It’s completely up to you whether an entry type should have a thumbnail, based on which field you select as the thumbnail source. That could be an Assets field, or any other field type which implements `ThumbableFieldInterface`.

What exactly would you expect to enter into a “media slot” beyond what’s already possible?

**Reply by thupsi** [reply]
I think @ryanleichty means something like this:

![Craft_cards_icon](https://github.com/user-attachments/assets/0cd0bdef-c52b-4287-a862-a8105d4e9d42)

Meaning, the entry type icon could act like a "fallback" for missing thumbnails. I agree it looks nicer.

Although, if the field/source contains multiple entry types, and some of them don't have an image neither an icon, then we would need an extra fallback? Maybe the first two letters of the entry type (like the user thumbnail)? 

Furthermore, if _none_ of the entry types in the field/source have images, they should be left alone, with no thumbnail IMO, because it takes up space.

Maybe this whole "fallbacks" thing should be configurable per entry type, or per field/source? 🙂

**Reply by ryanleichty** [reply]
@thupsi Yes, exactly like that! I meant to add a mockup but you nailed it.

Another option that would be nice that I don't think is possible currently would be using a thumbnail of a nested entry. Say you have an entry (a "section" in this scenario), and within that entry you have four nested entries ("cards") that each have an image which is set as the thumbnail. It would be cool if you could surface those nested entry thumbnails (using the first one available).

I'll have to see how Sanity handles the scenario if no image or icon is available.

**Reply by brandonkelly** [reply]
> Another option that would be nice that I don't think is possible currently would be using a thumbnail of a nested entry.

Just made that possible for Craft 5.5 (7fced5427b7a117c81d4a11d351a551c0a95b1d3).

**Reply by ryanleichty** [reply]
Awesome, thank you! 🔥

**Reply by ryanleichty** [reply]
> If the field/source contains multiple entry types, and some of them don't have an image neither an icon, then we would need an extra fallback?

In Sanity, they have a _system fallback icon_ if no custom icon or image is selected for the media slot, retaining the alignment of the items in the list.

![CleanShot 2024-09-04 at 08 48 50@2x](https://github.com/user-attachments/assets/2886bd41-907c-42b7-ad43-75213cdeb726)

I'm not sure how you would handle a fallback to a fallback in the UI, but for the general usage of the icon as thumbnail, perhaps adding a toggle in the entry type settings:

![CleanShot 2024-09-04 at 08 55 55@2x](https://github.com/user-attachments/assets/3e1ba402-d3b9-4308-8ed0-c08e35e7ef90)

The logic would follow something like this:

1. If a field has been set as the element thumbnail, use that
2. If no image is available in the selected field, or if no field has been set as the element thumbnail, and the icon is selected to be shown, show the icon
3. If no icon is selected???

**Reply by brandonkelly** [reply]
I like the idea of using the icon _if other entry types have thumbnails_. I think we would just continue not doing anything special for entry types that don’t have a thumbnail or an icon. The developer can go through and add the missing icons if it’s a problem.

**Reply by brandonkelly** [reply]
Craft 5.5.0 is out with the ability to select Entries fields as thumbnail providers 🎉

**Reply by ryanleichty** [reply]
Circling back to this after testing out the latest updates, and maybe I'm missing something, but I was expecting to see the icons as thumbnails here:

![CleanShot 2024-11-13 at 11 28 55@2x](https://github.com/user-attachments/assets/bd9ba0c7-1413-4d35-9a9c-59e64cad73f1)

Same thing here:

![CleanShot 2024-11-13 at 11 34 25@2x](https://github.com/user-attachments/assets/6fe7dcbe-6bc7-4990-8b4a-b3b93c1f44b3)

I do see the icon showed in the preview here, but not in the actual UI.

![CleanShot 2024-11-13 at 11 33 04@2x](https://github.com/user-attachments/assets/26aef24a-adef-4fb2-a666-6d3127afbc8d)

> Using a thumbnail of a nested entry

How is this supposed to work with matrix fields? Where do we select the thumbnail that should be used?

**Reply by thupsi** [reply]
I don't think the improvements in 5.5 have to do with the icon appearing in thumbnail form as described here. The kinda relevant new feature is that you can surface a thumbnail from an entry field to the parent entry.

**Reply by ryanleichty** [reply]
Yeah, I guess I took some of the responses above as confirmation that you could:

- Use a thumbnail from a nested entry
- Use an icon if other entry types have thumbnails

**Comment by brandonkelly** [comment, marked as answer]
This is sortof being addressed in Craft 5.7, as we’re moving the thumbnails over to the right side of the card, so they won’t look weird alongside cards without thumbs. (#16983)

**Reply by brandonkelly** [reply]
Craft 5.7.0 is out with that change.
